### ISSUE TITLE
Test build 2.2.20190125

### ISSUE THREAD

**Issue by BenRQ** [NPBruce/valkyrie#1013]
For those of you who want to test the latest unofficial build version 2.2 : 
Android: https://we.tl/t-fX7B6xJ0o5
Windows : https://we.tl/t-MW8tBjw8VZ
Mac: https://we.tl/t-AF9TQJ6vsQ

It includes all the development already done for 2.2, as all is now merged in the official dev branch.
Any issues detected in this version should be raised as a new issue here.

Link will expire in a few days : this is done on purpose :)

**Comment by scrubbless** [comment]
Testing for 2.2 - Android

Issues encountered:

- Import too 5-6 minutes

- When I loaded the scenario list it didn't display images (on poor 4g connection). Went back to main screen and it crashed Valkyrie. App froze up completely had to force quit.

- Missing text on content:
![screenshot_20190125-083105_valkyrie](https://user-images.githubusercontent.com/32058089/51734207-bd451c00-207b-11e9-99db-dd3db34d2e95.jpg)

- Home made scenarios (in your editor folder) no longer show. Means you can't test a scenario with specific expansions only selected.

- Exiting the scenario filter (author, rating etc) causes it to hang for a little while. Twice now I've tried to hit the 'Ok' button a second time (assume I miss pressed) and it started downloading a scenario.

- Missing reference names
![screenshot_20190125-083823_valkyrie](https://user-images.githubusercontent.com/32058089/51734526-b074f800-207c-11e9-996d-3e79bfd52b69.jpg)

![screenshot_20190125-083834_valkyrie](https://user-images.githubusercontent.com/32058089/51734550-b9fe6000-207c-11e9-9150-ca8e8a7a458e.jpg)

![screenshot_20190125-083937_valkyrie](https://user-images.githubusercontent.com/32058089/51734564-c5518b80-207c-11e9-8e1b-8bb9fbc1b2d7.jpg)

![screenshot_20190125-084007_valkyrie](https://user-images.githubusercontent.com/32058089/51734585-d9958880-207c-11e9-84d4-07bb4c5bdfec.jpg)


There are a lot of these.

Note: @BenRQ  during testing you should delete Valkyrie and the Valkyrie folder before installing and testing the new version. Import seems to be a very common issue.

**Comment by scrubbless** [comment]
Some of the issues also present on WIndows:

- Tile, Expansion, Investigator Reference links are broken:
![image](https://user-images.githubusercontent.com/32058089/51737099-ac000d80-2083-11e9-8741-749c551e5a0b.png)
![image](https://user-images.githubusercontent.com/32058089/51737141-c9cd7280-2083-11e9-9f0d-e832adfc9659.png)
![image](https://user-images.githubusercontent.com/32058089/51737152-d2be4400-2083-11e9-807f-d776a18a46d4.png)
![image](https://user-images.githubusercontent.com/32058089/51737187-e23d8d00-2083-11e9-8955-204fa9cd87d9.png)

- Home Made scenarios not showing in new scenario list.
![image](https://user-images.githubusercontent.com/32058089/51737957-e074c900-2085-11e9-9057-6a2aa064c81e.png)

(Its not there :()
![image](https://user-images.githubusercontent.com/32058089/51737976-eff41200-2085-11e9-85fb-ab3caa096cb8.png)

**Comment by BenRQ** [comment]
>Tile, Expansion, Investigator Reference links are broken:

OK I will have a look at the missing reference. It should be an issue with import.

> Home made scenarios (in your editor folder) no longer show. Means you can't test a scenario with specific expansions only selected.

Yes, I got a lot of invalid test report from people testing their scenario at the beginning of the rating activation.
I had to make sure scenario starting with "EditorScenario" would not send their feedback which is problematic for some official scenario.
So I would prefer to implement this : #511 Control loaded content in test function
And keep the local scenario out of the "online" quest mode.

> Offline mode does not allow you to play any downloaded scenarios (just has the download arrow and then goes to a "Downloading package" screen that hangs.

OK that's a bug

> My client for some reason now appears to be stuck in online mode, I have an internet connection (I actually have a remote desktop sessions to it, to use it... so definatly an internet connection there). It happened after I created a new scenario and restarted Valkyrie.

Strange ... do you have a way to reproduce this again so I can try it ?

**Comment by scrubbless** [comment]
> Yes, I got a lot of invalid test report from people testing their scenario at the beginning of the rating activation.
> I had to make sure scenario starting with "EditorScenario" would not send their feedback which is problematic for some official scenario.
> So I would prefer to implement this : #511 Control loaded content in test function
> And keep the local scenario out of the "online" quest mode.

Its very important to be able to test a scenario "as a play would play it". The Save& Test in the scenario screen automatically enables all expansions. So you can't truly test.

> Strange ... do you have a way to reproduce this again so I can try it ?

I will see if I can find steps to reproduce.

**Comment by BenRQ** [comment]
> The Save& Test in the scenario screen automatically enables all expansions. So you can't truly test.

Purpose of #511 is to allow dynamic choice of expansions

**Comment by redwolf2** [comment]
I had the same problem as @scrubbless with the texts on the pc platform. {ffg: texts missing, even though they are confirmed available in the import files. 2.1.5 import works.

**Comment by scrubbless** [comment]
We still need an option to "play as the user will play" before we publish our scenarios.

For example, I have events in A Strain on Reality. The expansions required are "None". But I have events I need to test when certain expansions are in a players content library.

For example:
![image](https://user-images.githubusercontent.com/32058089/51747926-c5648200-20a2-11e9-8bd9-d53b2563cb55.png)
(no expansions 'Required')

An Event that looks to see if the player has Streets of Arkham, then runs if they do:
![image](https://user-images.githubusercontent.com/32058089/51748033-12485880-20a3-11e9-8c13-
a9ba377eb55f.png)
![image](https://user-images.githubusercontent.com/32058089/51748055-22f8ce80-20a3-11e9-8c03-76cad438fd75.png)

As you can see I need the ability to pick which expansions are in my library when testing the scenario.

**Comment by BenRQ** [comment]
Yes that's what I meant, with #511 implemented you should be able to select expansion as you'd like in the editor for testing in the editor, with save & test.


That's my understanding of the ticket at least.

**Comment by BenRQ** [comment]
I need to have a look at these import issues but I won't be able to do it this weekend.
In the meantime you should be able to import from version 2.1.5 and run 2.2 without importing again.

**Comment by chrismaliszewski** [comment]
It's not a bug but I don't like the feature of clicking a scenario name causing its downloading. I downloaded two scenarios although I didn't want them by simply mistakenly clicking a scenario line. To me, the concept is wrong and bad! 
I think, that downloading a scenario should only happen when you click a download button. 
Same goes with starting it, with the button which appears instead of the download one. 
When you click a scenario line, either nothing should happen OR a scenario's title screen with its description should appear and in here you either should be able to Start a scenario (if you've downloaded it already) or download it if you want as you know what this scenario is about.

**Comment by chrismaliszewski** [comment]
There are **many** filtering problems.

Sorting doesn't work in Polish properly. Choose any type of sorting! 
Example: sort by Name ascending. For some reason, 'En la pajareria...' is before 'A glimpse of the Past' and there are a few other mistakes in the ordering. 

Problem with Difficulty sorting in EN and PL. 

Finally, hiding scenarios with missing expansions doesn't work either. I have no expansions selected but scenarios are still visible.

**Comment by BenRQ** [comment]
@chrismaliszewski you have sortering issue because you changed languages while testing.
Restart in Polish and you won't have problems.
This is a bug of course I will fix this.

For the filters, I don't understand the problem. All the scenarios that do not require expansions will be displayed.
Please share screenshot if you see something unexpected.

**Comment by BenRQ** [comment]
@chrismaliszewski clicking the scenario to download is the current behaviour of official version.
I like this behaviour, but I can change that if everyone think this is a better idea to limit the action to the button.

**Comment by scrubbless** [comment]
With the menu lag, I think clicking the scenario text is annoying. 


If its possible opening the scenarios splash age would be better. However I suspect that will require downloading the whole scenario.

**Comment by chrismaliszewski** [comment]
@scrubbless, @BenRQ - I created a new issue relating the Scenario explorer matter, here: https://github.com/NPBruce/valkyrie/issues/1020

**Comment by alvaromunoz** [comment]
Testing for MacOS (10.14.2 Mojave), MoM and DrtL at their latest versions according to Steam.

The Beta (2.2.20190125) seems to freeze after the "Made with unity" logo. I've left it running for 5 minutes and no response so far:
<img width="532" alt="captura de pantalla 2019-01-27 a la s 2 59 25 p m" src="https://user-images.githubusercontent.com/2722773/51804837-37c09800-2244-11e9-84d7-440fe5faed8e.png">

For what it's worth, the official release (2.1.5) does work fine:
<img width="688" alt="captura de pantalla 2019-01-27 a la s 3 01 57 p m" src="https://user-images.githubusercontent.com/2722773/51804868-80785100-2244-11e9-9715-cd912c9710b8.png">

Let me know if you need any logs from anywhere on the system...

**Comment by scrubbless** [comment]
There is a log here:

%LOCALAPPDATA%Low\NA\Valkyrie\output_log.txt

**Comment by chrismaliszewski** [comment]
Another issue two new issues found. 

1. For some reason when I opened the valkyrie-windows-2.2.20190125 it went offline although I am online. But the bug is: when you click Start Scenario you see Offline caption; then you come back to the menu and again Start Scenario and the caption disappears although you are still in Offline mode. At least in my case. See the attachment below without 'Offline' caption.

2. I think it might be related to the 1.. I downloaded 1 scenario for testing valkyrie-windows-2.2.20190125. Still in offline mode. Although the app says it has filtered out 5 scenarios (I have no idea how 5...), there is nothing displayed. 
![clipboard01](https://user-images.githubusercontent.com/7532602/51864660-378ddd00-23a9-11e9-9927-241b206638e2.jpg)

**Comment by redwolf2** [comment]
I added them as separate issues

**Comment by chrismaliszewski** [comment]
As I mentioned in a few other issues and @scrubbless in his comment https://github.com/NPBruce/valkyrie/issues/1013#issuecomment-457515661 (last dot), the app can switch to 'forever offline mode' and there is no way to make it works again. 

I tried:
1. removing Valkyrie dir in %appdata% and reinitiating it (no success),
2. removing valkyrie-windows-2.2.201901225 catalogue and unzipping it again under a new path (no success),
3. removing %userprofile%\appdata\locallow\NA\Valkyrie (no success),
4. adding the app to firewall exceptions (just in case) (no success),
5. playing with regedit. No success. I removed NA/Valkyrie subkey from it, run the app and it was online. So I restarted the app but it was offline again and removing the subkey doesn't help again. So it was something else but I don't know what exactly.
